docs/ugcstor-csi.md

```diff
@@ -0,0 +1,802 @@
+```
+Sample Output:
+```
+NAME                   HEALTHYINSTANCES   PROVISIONEDINSTANCES   DESIREDINSTANCES     AGE
+cstor-disk-pool        1                  1                      1                    2m2s
+```
+Check if the pool instances report their status as <b>ONLINE</b> using the below command:
+
+```
+kubectl get cspi -n openebs
+```
+Sample Output:
+
+```
+NAME                  HOSTNAME             ALLOCATED   FREE    CAPACITY   STATUS   AGE
+cstor-disk-pool-vn92  worker-node-1        60k         9900M    9900M     ONLINE   2m17s
+cstor-disk-pool-al65  worker-node-2        60k         9900M    9900M     ONLINE   2m17s
+cstor-disk-pool-y7pn  worker-node-3        60k         9900M    9900M     ONLINE   2m17s
+```
+Once all the pods are in running state, these pool instances can be used for creation of cStor volumes.
+<hr>
+
+### <a class="anchor" aria-hidden="true" id="creating-cstor-storage-classes"></a>Creating cStor storage classes
+
+StorageClass definition is an important task in the planning and execution of OpenEBS storage. The real power of CAS architecture is to give an independent or a dedicated storage engine like cStor for each workload, so that granular policies can be applied to that storage engine to tune the behaviour or performance as per the workload's need.
+  
+  #### Steps to create a cStor StorageClass:
+   1. Decide the CStorPoolCluster for which you want to create a Storage Class.
+   2. Decide the replicaCount based on your requirement/workloads. OpenEBS doesn't restrict the replica count to set, but a <b>maximum of 5</b> replicas are allowed. It depends how users configure it, but for the availability of volumes <b>at least (n/2 + 1) replicas</b> should be up and connected to the target, where n is the replicaCount. The Replica Count should be greater than or equal to the number of cStor Pool Instances(CSPIs). The following are some example cases:
+    <ul> 
+    <li>If a user configured replica count as 2, then always 2 replicas should be available to perform operations on volume.</li>
+    <li>If a user configured replica count as 3 it should require at least 2 replicas should be available for volume to be operational.</li>
+     <li>If a user configured replica count as 5 it should require at least 3 replicas should be available for volume to be operational.</li>
+    </ul>
+   3. Create a YAML spec file <code>cstor-csi-disk.yaml</code> using the template given below. Update the pool, replica count and other policies. By using this sample configuration YAML, a StorageClass will be created with 3 OpenEBS cStor replicas and will configure themselves on the pool instances. 
+
+      ```
+      kind: StorageClass
+      apiVersion: storage.k8s.io/v1
+      metadata:
+        name: cstor-csi-disk
+      provisioner: cstor.csi.openebs.io
+      allowVolumeExpansion: true
+      parameters:
+        cas-type: cstor
+        # cstorPoolCluster should have the name of the CSPC
+        cstorPoolCluster: cstor-storage
```

Comment:
To be intact let's update the value to `cstor-disk-pool`(Since CSPC is created with the same name).